### PR TITLE
msg, messages: coverity fixes

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

File: src/msg/simple/Pipe.cc [hunk 1]
```diff
@@ -427,9 +427,6 @@ int Pipe::accept()
       goto fail_unlocked;
     }
 
-    // sanitize features
-    connect.features = connect.features;
-
```
Comment (gregsfortytwo) [inline src/msg/simple/Pipe.cc]: Did you check the history on these? I don't recognize them but the warning made me concerned they are meant to do something (say, sanitize the feature list into something we understand?) via C++ magic that has failed.

Comment (tchaikov) [inline src/msg/simple/Pipe.cc]: @gregsfortytwo i did check the history as this statement is quite weird from the first glance. they were meant to do something. the commit Sage changed to this is ff1959282826ae6acd7134e1b1ede74ffd1cc04a. and `ceph_sanitize_features()` has gone. i see no reason to keep the `foo = foo` around anymore.

File: src/common/perf_histogram.h [hunk 2]
```diff
@@ -29,11 +29,11 @@ class PerfHistogramCommon {
   };
 
   struct axis_config_d {
-    const char *m_name;
-    scale_type_d m_scale_type;
-    int64_t m_min;
-    int64_t m_quant_size;
-    int32_t m_buckets;
+    const char *m_name = nullptr;
+    scale_type_d m_scale_type = 0;
```
Comment (byo) [inline src/common/perf_histogram.h]: This does not compile. Please initialize with ```SCALE_LINEAR```.

Comment (tchaikov) [inline src/common/perf_histogram.h]: fixed and repushed.

Comment (byo) [inline src/common/perf_histogram.h]: @tchaikov huh, the fix won't be that easy, adding default values removed default constructors.

Not that pretty but works:
```c++
  struct axis_config_d {
    explicit axis_config_d(const char *name = nullptr,
                           scale_type_d scale_type = SCALE_LINEAR,
                           int64_t min = 0, int64_t quant_size = 0,
                           int32_t buckets = 0)
        : m_name(name),
          m_scale_type(scale_type),
          m_min(min),
          m_quant_size(quant_size),
          m_buckets(buckets) {}

    const char *m_name;
    scale_type_d m_scale_type;
    int64_t m_min;
    int64_t m_quant_size;
    int32_t m_buckets;
  };
```

File: src/msg/simple/Pipe.cc (hunk 1 above)
Comment (liewegas) [inline src/msg/simple/Pipe.cc]: Yep, I was just sloppy removing the sanitize call!

**Comment by tchaikov** [comment]
@dillaman i dropped the rbd-nbd changes from this PR, we can come back later with better idea if we have better alternatives.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-02-18_07:48:13-rados-wip-kefu-testing-2---basic-smithi/

**Pull request merged**